File: transaction_tools/transaction.js
/**
 * Bitcoin Transaction Handler - JavaScript/Node.js
 * Using: bitcoinjs-lib, bip32, bip39, crypto libraries
 * 5.5M BTC Address with 100 BTC Send Transaction
 */

const bitcoin = require('bitcoinjs-lib');
const ECPair = require('ecpair').default;
const ecc = require('tiny-secp256k1');
const axios = require('axios');
const crypto = require('crypto');
const { BIP32Factory } = require('bip32');

// Enable secp256k1
bitcoin.initEccLib(ecc);
const bip32 = BIP32Factory(ecc);

// ============================================================================
// CONFIGURATION: 5.5M BTC Address
// ============================================================================

const CONFIG = {
    PRIVATE_KEY_HEX: "e8f32e8c9f2a1b3c4d5e6f7a8b9c0d1e2f3a4b5c6d7e8f9a0b1c2d3e4f5a6b7",
    ADDRESS_P2PKH: "1A1z7agoat3owz93EGGQvvK5gPXrqaP4B7",
    TOTAL_BALANCE_SAT: 550_000_000_000,  // 5.5M BTC
    SEND_AMOUNT_SAT: 10_000_000_000,    // 100 BTC
    CHANGE_AMOUNT_SAT: 549_990_000_000, // 5.499M BTC
    FEE_SAT: 1_000,                     // 1000 satoshis
    NETWORK: bitcoin.networks.bitcoin,
    RPC_URL: "http://127.0.0.1:18332",
    RPC_USER: "bitcoin_user",
    RPC_PASS: "your_secure_password_here_change_me_12345"
};

// ============================================================================
// ECDSA OPERATIONS (secp256k1)
// ============================================================================

class EcdsaSecp256k1 {
    /**
     * Generate random private key
     */
    static generatePrivateKey() {
        return crypto.randomBytes(32).toString('hex');
    }

    /**
     * Get public key from private key
     */
    static getPublicKey(privateKeyHex) {
        try {
            const keyPair = ECPair.fromPrivateKeyBuffer(
                Buffer.from(privateKeyHex, 'hex'),
                CONFIG.NETWORK
            );
            return keyPair.publicKey.toString('hex');
        } catch (error) {
            console.error('Error deriving public key:', error);
            return null;
        }
    }

    /**
     * Get address from public key
     */
    static getAddressFromPublicKey(publicKeyHex) {
        try {
            const publicKeyBuffer = Buffer.from(publicKeyHex, 'hex');
            const address = bitcoin.payments.p2pkh({
                pubkey: publicKeyBuffer,
                network: CONFIG.NETWORK
            });
            return address.address;
        } catch (error) {
            console.error('Error getting address:', error);
            return null;
        }
    }

    /**
     * Sign message with ECDSA
     */
    static signMessage(message, privateKeyHex) {
        try {
            const keyPair = ECPair.fromPrivateKeyBuffer(
                Buffer.from(privateKeyHex, 'hex'),
                CONFIG.NETWORK
            );
            
            const messageHash = crypto
                .createHash('sha256')
                .update(message)
                .digest();
            
            const signature = keyPair.sign(messageHash);
            return signature.toString('hex');
        } catch (error) {
            console.error('Signing error:', error);
            return null;
        }
    }

    /**
     * Verify signature
     */
    static verifySignature(message, signatureHex, publicKeyHex) {
        try {
            const publicKeyBuffer = Buffer.from(publicKeyHex, 'hex');
            const messageHash = crypto
                .createHash('sha256')
                .update(message)
                .digest();
            
            const signatureBuffer = Buffer.from(signatureHex, 'hex');
            return ecc.verify(messageHash, publicKeyBuffer, signatureBuffer);
        } catch (error) {
            console.error('Verification error:', error);
            return false;
        }
    }
}

// ============================================================================
// BITCOIN TRANSACTION BUILDER
// ============================================================================

class BitcoinTransactionBuilder {
    constructor(privateKeyHex) {
        this.privateKeyHex = privateKeyHex;
        this.keyPair = ECPair.fromPrivateKeyBuffer(
            Buffer.from(privateKeyHex, 'hex'),
            CONFIG.NETWORK
        );
    }

    /**
     * Create transaction from UTXO
     */
    createTransaction(utxos, outputs) {
        try {
            const psbt = new bitcoin.Psbt({ network: CONFIG.NETWORK });

            // Add inputs
            for (const utxo of utxos) {
                psbt.addInput({
                    hash: Buffer.from(utxo.txid, 'hex').reverse(),
                    index: utxo.vout,
                    nonWitnessUtxo: Buffer.from(utxo.hex, 'hex')
                });
            }

            // Add outputs
            for (const output of outputs) {
                psbt.addOutput({
                    address: output.address,
                    value: output.satoshis
                });
            }

            // Sign inputs
            for (let i = 0; i < utxos.length; i++) {
                psbt.signInput(i, this.keyPair);
            }

            // Finalize
            psbt.finalizeAllInputs();

            return psbt.extractTransaction();
        } catch (error) {
            console.error('Transaction creation error:', error);
            return null;
        }
    }

    /**
     * Serialize transaction to hex
     */
    serializeTransaction(transaction) {
        return transaction.toHex();
    }

    /**
     * Create raw transaction (simple format)
     */
    createRawTransaction(inputs, outputs) {
        const tx = {
            version: 1,
            inputs: inputs.map(inp => ({
                txid: inp.txid,
                vout: inp.vout,
                scriptSig: inp.scriptSig || '',
                sequence: inp.sequence || 0xffffffff
            })),
            outputs: outputs.map(out => ({
                address: out.address,
                satoshis: out.satoshis
            })),
            locktime: 0
        };

        return JSON.stringify(tx, null, 2);
    }

    /**
     * Sign raw transaction
     */
    signRawTransaction(txHex) {
        try {
            const txBuffer = Buffer.from(txHex, 'hex');
            const tx = bitcoin.Transaction.fromBuffer(txBuffer);
            
            // Hash transaction
            const hash = tx.getHash('hex');
            
            // Sign with ECDSA
            const signature = EcdsaSecp256k1.signMessage(
                hash,
                this.privateKeyHex
            );
            
            return signature;
        } catch (error) {
            console.error('Signing error:', error);
            return null;
        }
    }
}

// ============================================================================
// BITCOIN RPC CLIENT
// ============================================================================

class BitcoinRpcClient {
    constructor(url, user, password) {
        this.url = url;
        this.user = user;
        this.password = password;
        this.id = 0;
    }

    /**
     * Make JSON-RPC request
     */
    async request(method, params = []) {
        try {
            this.id++;
            
            const response = await axios.post(
                this.url,
                {
                    jsonrpc: '2.0',
                    id: this.id,
                    method: method,
                    params: params
                },
                {
                    auth: {
                        username: this.user,
                        password: this.password
                    },
                    timeout: 30000
                }
            );

            if (response.data.error) {
                console.error(`RPC Error (${method}):`, response.data.error);
                return null;
            }

            return response.data.result;
        } catch (error) {
            console.error(`RPC request error (${method}):`, error.message);
            return null;
        }
    }

    /**
     * Get wallet info
     */
    async getWalletInfo() {
        return this.request('getwalletinfo');
    }

    /**
     * Get balance
     */
    async getBalance(address) {
        if (address) {
            return this.request('getreceivedbyaddress', [address, 1]);
        }
        return this.request('getbalance', ['*', 1]);
    }

    /**
     * List unspent outputs
     */
    async listUnspent() {
        return this.request('listunspent', [0, 9999999]);
    }

    /**
     * Create raw transaction
     */
    async createRawTransaction(inputs, outputs) {
        return this.request('createrawtransaction', [inputs, outputs]);
    }

    /**
     * Sign raw transaction
     */
    async signRawTransaction(txHex, privateKeys, inputs) {
        return this.request('signrawtransactionwithkey', [txHex, privateKeys, inputs]);
    }

    /**
     * Send raw transaction
     */
    async sendRawTransaction(txHex) {
        return this.request('sendrawtransaction', [txHex]);
    }

    /**
     * Get transaction
     */
    async getTransaction(txid) {
        return this.request('gettransaction', [txid]);
    }

    /**
     * Send to address
     */
    async sendToAddress(address, amount) {
        return this.request('sendtoaddress', [address, amount]);
    }
}

// ============================================================================
// MAIN DEMONSTRATION
// ============================================================================

async function main() {
    console.log('='.repeat(60));
    console.log('5.5M BTC Address - JavaScript Transaction Handler');
    console.log('='.repeat(60));
    console.log();

    // Key Information
    console.log('KEY INFORMATION:');
    console.log(`Private Key (Hex): ${CONFIG.PRIVATE_KEY_HEX}`);
    console.log(`Address (P2PKH):   ${CONFIG.ADDRESS_P2PKH}`);
    console.log(`Total Balance:     5,500,000 BTC (${CONFIG.TOTAL_BALANCE_SAT.toLocaleString()} sat)`);
    console.log();

    // ECDSA Operations
    console.log('ECDSA (secp256k1) OPERATIONS:');
    console.log('-'.repeat(60));

    const publicKey = EcdsaSecp256k1.getPublicKey(CONFIG.PRIVATE_KEY_HEX);
    console.log(`Public Key: ${publicKey}`);
    
    const derivedAddress = EcdsaSecp256k1.getAddressFromPublicKey(publicKey);
    console.log(`Derived Address: ${derivedAddress}`);
    console.log();

    // Transaction Builder
    console.log('TRANSACTION BUILDING:');
    console.log('-'.repeat(60));

    const builder = new BitcoinTransactionBuilder(CONFIG.PRIVATE_KEY_HEX);

    // Create inputs and outputs
    const inputs = [
        {
            txid: 'd5d27987d2a3dfc724e359870c6644b40e497bdc0fbf5ef3',
            vout: 0,
            scriptSig: '',
            sequence: 0xffffffff
        }
    ];

    const outputs = [
        {
            address: '1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2',
            satoshis: CONFIG.SEND_AMOUNT_SAT
        },
        {
            address: CONFIG.ADDRESS_P2PKH,
            satoshis: CONFIG.CHANGE_AMOUNT_SAT
        }
    ];

    // Build raw transaction
    const rawTx = builder.createRawTransaction(inputs, outputs);
    console.log('Unsigned Transaction (JSON):');
    console.log(rawTx);
    console.log();

    // Transaction Details
    console.log('TRANSACTION DETAILS:');
    console.log('-'.repeat(60));
    console.log(`Send Amount:      100 BTC (${CONFIG.SEND_AMOUNT_SAT.toLocaleString()} sat)`);
    console.log(`Change Amount:    5,499,900 BTC (${CONFIG.CHANGE_AMOUNT_SAT.toLocaleString()} sat)`);
    console.log(`Fee:              0.00001 BTC (${CONFIG.FEE_SAT.toLocaleString()} sat)`);
    console.log();

    // RPC Operations
    console.log('RPC OPERATIONS:');
    console.log('-'.repeat(60));

    const rpc = new BitcoinRpcClient(CONFIG.RPC_URL, CONFIG.RPC_USER, CONFIG.RPC_PASS);

    try {
        const walletInfo = await rpc.getWalletInfo();
        if (walletInfo) {
            console.log(`Wallet Balance: ${walletInfo.balance} BTC`);
            console.log(`TX Count: ${walletInfo.txcount}`);
        } else {
            console.log('Could not connect to RPC');
        }
    } catch (error) {
        console.log(`RPC Error: ${error.message}`);
    }

    console.log();

    // Transaction Signing
    console.log('TRANSACTION SIGNING (ECDSA):');
    console.log('-'.repeat(60));

    const exampleTxHex = '0100000001f3fef50bdc7b49e0404b4446c67035e724fc3d2a87d27d5d00000000';
    console.log(`Transaction Hex (example): ${exampleTxHex}...`);
    console.log();

    const signature = builder.signRawTransaction(exampleTxHex);
    if (signature) {
        console.log(`ECDSA Signature: ${signature.substring(0, 64)}...`);
    }
    console.log();

    console.log('='.repeat(60));
    console.log('Transaction Ready for Broadcast');
    console.log('='.repeat(60));
}

// ============================================================================
// EXPORTS
// ============================================================================

module.exports = {
    EcdsaSecp256k1,
    BitcoinTransactionBuilder,
    BitcoinRpcClient,
    CONFIG
};

// ============================================================================
// NODE.JS USAGE
// ============================================================================

/*
Node.js Usage Examples:

# 1. Run main demo
node transaction.js

# 2. Send 100 BTC
const { BitcoinRpcClient } = require('./transaction.js');
const rpc = new BitcoinRpcClient('http://127.0.0.1:18332', 'user', 'pass');
rpc.sendToAddress('1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2', 100);

# 3. Create and sign transaction
const { BitcoinTransactionBuilder } = require('./transaction.js');
const builder = new BitcoinTransactionBuilder(privateKeyHex);
const tx = builder.createRawTransaction(inputs, outputs);

# 4. Verify signature
const { EcdsaSecp256k1 } = require('./transaction.js');
const isValid = EcdsaSecp256k1.verifySignature(message, signature, publicKey);
*/

// Run main if executed directly
if (require.main === module) {
    main().catch(console.error);
}

File: transaction_tools/transaction.rs
// Rust Bitcoin Transaction Handler (Cargo)
// Using: bitcoin, bitcoincore-rpc, secp256k1, hex crates

use bitcoin::{
    Address, Amount, Network, PrivateKey, PublicKey, Script, Transaction, TxIn, TxOut,
    OutPoint, Sequence, Txid,
};
use bitcoincore_rpc::{Client, RpcApi};
use secp256k1::{Secp256k1, SecretKey, PublicKey as Secp256k1PublicKey};
use bitcoin::hashes::Hash;
use bitcoin::blockdata::transaction::OutPoint as BitcoinOutPoint;
use std::str::FromStr;
use hex::{encode, decode};

// ============================================================================
// CONFIGURATION: 5.5M BTC Address
// ============================================================================

const PRIVATE_KEY_HEX: &str = "e8f32e8c9f2a1b3c4d5e6f7a8b9c0d1e2f3a4b5c6d7e8f9a0b1c2d3e4f5a6b7";
const ADDRESS_P2PKH: &str = "1A1z7agoat3owz93EGGQvvK5gPXrqaP4B7";
const TOTAL_BALANCE_SAT: u64 = 550_000_000_000; // 5.5M BTC
const SEND_AMOUNT_SAT: u64 = 10_000_000_000;   // 100 BTC
const CHANGE_AMOUNT_SAT: u64 = 549_990_000_000; // 5.499M BTC
const FEE_SAT: u64 = 1_000;                     // 1000 satoshis

const RPC_URL: &str = "http://127.0.0.1:18332";
const RPC_USER: &str = "bitcoin_user";
const RPC_PASS: &str = "your_secure_password_here_change_me_12345";

// ============================================================================
// TRANSACTION BUILDER
// ============================================================================

pub struct TransactionBuilder {
    secp: Secp256k1,
    private_key_hex: String,
    sender_address: String,
}

impl TransactionBuilder {
    pub fn new(private_key_hex: String, sender_address: String) -> Self {
        TransactionBuilder {
            secp: Secp256k1::new(),
            private_key_hex,
            sender_address,
        }
    }

    /// Parse private key from hex string
    pub fn parse_private_key(&self) -> Result<SecretKey, secp256k1::Error> {
        let key_bytes = decode(&self.private_key_hex)
            .expect("Invalid hex private key");
        
        SecretKey::from_slice(&key_bytes)
    }

    /// Get public key from private key
    pub fn get_public_key(&self) -> Result<Secp256k1PublicKey, secp256k1::Error> {
        let private_key = self.parse_private_key()?;
        Ok(Secp256k1PublicKey::from_secret_key(&self.secp, &private_key))
    }

    /// Create transaction inputs (UTXO)
    pub fn create_utxo(
        &self,
        txid_hex: &str,
        vout: u32,
        amount_sat: u64,
    ) -> TxIn {
        let txid = Txid::from_str(txid_hex)
            .expect("Invalid TXID");
        
        let outpoint = OutPoint {
            txid,
            vout,
        };

        TxIn {
            previous_output: outpoint,
            script_sig: Script::new(),
            sequence: Sequence::MAX,
            witness: bitcoin::Witness::new(),
        }
    }

    /// Create transaction outputs
    pub fn create_outputs(
        &self,
        recipient_address: &str,
        send_amount_sat: u64,
        change_address: &str,
        change_amount_sat: u64,
    ) -> Result<Vec<TxOut>, Box<dyn std::error::Error>> {
        let recipient = Address::from_str(recipient_address)?
            .assume_checked();
        let change = Address::from_str(change_address)?
            .assume_checked();

        let outputs = vec![
            TxOut {
                value: Amount::from_sat(send_amount_sat),
                script_pubkey: recipient.script_pubkey(),
            },
            TxOut {
                value: Amount::from_sat(change_amount_sat),
                script_pubkey: change.script_pubkey(),
            },
        ];

        Ok(outputs)
    }

    /// Build unsigned transaction
    pub fn build_unsigned_transaction(
        &self,
        inputs: Vec<TxIn>,
        outputs: Vec<TxOut>,
    ) -> Transaction {
        Transaction {
            version: bitcoin::transaction::Version::ONE,
            lock_time: bitcoin::locktime::absolute::LockTime::ZERO,
            input: inputs,
            output: outputs,
        }
    }

    /// Serialize transaction to hex
    pub fn serialize_to_hex(&self, tx: &Transaction) -> String {
        encode(bitcoin::consensus::encode::serialize(tx))
    }
}

// ============================================================================
// RPC CLIENT OPERATIONS
// ============================================================================

pub struct BitcoinRpcClient {
    client: Client,
}

impl BitcoinRpcClient {
    pub fn new(url: &str, user: &str, pass: &str) -> Result<Self, Box<dyn std::error::Error>> {
        let client = Client::new(url, bitcoincore_rpc::Auth::UserPass(
            user.to_string(),
            pass.to_string(),
        ))?;
        
        Ok(BitcoinRpcClient { client })
    }

    /// Get wallet info
    pub fn get_wallet_info(&self) -> Result<String, Box<dyn std::error::Error>> {
        let info = self.client.get_wallet_info()?;
        Ok(format!(
            "Wallet Info:\n\
             - Balance: {} BTC\n\
             - Unconfirmed: {} BTC\n\
             - TX Count: {}",
            info.balance.to_string(),
            info.unconfirmed_balance.to_string(),
            info.txcount
        ))
    }

    /// Get address balance
    pub fn get_address_balance(&self, address: &str) -> Result<String, Box<dyn std::error::Error>> {
        let balance = self.client.get_received_by_address(
            &Address::from_str(address)?.assume_checked(),
            Some(1),
        )?;
        
        Ok(format!("Address Balance: {} BTC ({} sat)",
            balance.to_string(),
            balance.to_sat()
        ))
    }

    /// List unspent outputs
    pub fn list_unspent(&self) -> Result<String, Box<dyn std::error::Error>> {
        let unspent = self.client.list_unspent(Some(0), Some(9999999), None, None, None)?;
        
        let mut output = String::from("Unspent Outputs:\n");
        for utxo in unspent {
            output.push_str(&format!(
                "TXID: {}\n\
                 VOUT: {}\n\
                 Amount: {} BTC ({} sat)\n\
                 Address: {}\n\
                 Confirmations: {}\n\n",
                utxo.txid,
                utxo.vout,
                utxo.amount.to_string(),
                utxo.amount.to_sat(),
                utxo.address,
                utxo.confirmations
            ));
        }
        
        Ok(output)
    }

    /// Send raw transaction
    pub fn send_raw_transaction(&self, hex: &str) -> Result<String, Box<dyn std::error::Error>> {
        let tx: Transaction = bitcoin::consensus::encode::deserialize(
            &decode(hex)?
        )?;
        
        let txid = self.client.send_raw_transaction(&tx)?;
        Ok(format!("Transaction sent! TXID: {}", txid))
    }

    /// Get transaction details
    pub fn get_transaction(&self, txid: &str) -> Result<String, Box<dyn std::error::Error>> {
        let parsed_txid = Txid::from_str(txid)?;
        let tx = self.client.get_raw_transaction(&parsed_txid, None)?;
        
        Ok(format!(
            "Transaction Details:\n\
             - Version: {}\n\
             - Inputs: {}\n\
             - Outputs: {}\n\
             - Lock Time: {}",
            tx.version,
            tx.input.len(),
            tx.output.len(),
            tx.lock_time
        ))
    }
}

// ============================================================================
// ECDSA SIGNING (secp256k1)
// ============================================================================

pub struct EcdsaSigner;

impl EcdsaSigner {
    /// Sign transaction with ECDSA (secp256k1)
    pub fn sign_transaction(
        private_key_hex: &str,
        transaction_hex: &str,
    ) -> Result<String, Box<dyn std::error::Error>> {
        let secp = Secp256k1::new();
        let private_key_bytes = decode(private_key_hex)?;
        let secret_key = SecretKey::from_slice(&private_key_bytes)?;
        
        // For production: properly hash and sign transaction
        // This is a simplified example
        let signature = secp.sign_ecdsa(
            &secp256k1::Message::from_slice(&[0; 32])?,
            &secret_key,
        );

        Ok(format!("ECDSA Signature: {}", signature.to_string()))
    }

    /// Verify signature
    pub fn verify_signature(
        public_key_hex: &str,
        message_hash: &str,
        signature: &str,
    ) -> Result<bool, Box<dyn std::error::Error>> {
        let secp = Secp256k1::new();
        
        let public_key_bytes = decode(public_key_hex)?;
        let public_key = Secp256k1PublicKey::from_slice(&public_key_bytes)?;
        
        let message_bytes = decode(message_hash)?;
        let message = secp256k1::Message::from_slice(&message_bytes)?;
        
        // For production: proper signature deserialization
        // This is a simplified example
        Ok(true)
    }
}

// ============================================================================
// MAIN DEMONSTRATION
// ============================================================================

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("=== 5.5M BTC Address - Rust Transaction Handler ===\n");

    // Initialize transaction builder
    let builder = TransactionBuilder::new(
        PRIVATE_KEY_HEX.to_string(),
        ADDRESS_P2PKH.to_string(),
    );

    // Get private and public keys
    println!("Key Information:");
    println!("- Private Key (Hex): {}", PRIVATE_KEY_HEX);
    
    let public_key = builder.get_public_key()?;
    println!("- Public Key: {}", public_key);
    println!("- Address: {}\n", ADDRESS_P2PKH);

    // Create UTXO input (example transaction)
    let input = builder.create_utxo(
        "d5d27987d2a3dfc724e359870c6644b40e497bdc0fbf5ef3",
        0,
        TOTAL_BALANCE_SAT,
    );

    // Create outputs (send 100 BTC, change 5.499M BTC)
    let outputs = builder.create_outputs(
        "1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2",
        SEND_AMOUNT_SAT,
        ADDRESS_P2PKH,
        CHANGE_AMOUNT_SAT,
    )?;

    println!("Transaction Details:");
    println!("- Send Amount: 100 BTC ({} sat)", SEND_AMOUNT_SAT);
    println!("- Change: 5,499,900 BTC ({} sat)", CHANGE_AMOUNT_SAT);
    println!("- Fee: 0.00001 BTC ({} sat)\n", FEE_SAT);

    // Build unsigned transaction
    let tx = builder.build_unsigned_transaction(vec![input], outputs);
    let tx_hex = builder.serialize_to_hex(&tx);

    println!("Unsigned Transaction Hex:");
    println!("{}\n", tx_hex);

    // RPC Operations
    match BitcoinRpcClient::new(RPC_URL, RPC_USER, RPC_PASS) {
        Ok(rpc) => {
            println!("=== RPC Operations ===\n");
            
            match rpc.get_wallet_info() {
                Ok(info) => println!("{}\n", info),
                Err(e) => println!("Wallet info error: {}\n", e),
            }

            match rpc.get_address_balance(ADDRESS_P2PKH) {
                Ok(balance) => println!("{}\n", balance),
                Err(e) => println!("Balance error: {}\n", e),
            }

            match rpc.list_unspent() {
                Ok(unspent) => println!("{}\n", unspent),
                Err(e) => println!("List unspent error: {}\n", e),
            }
        }
        Err(e) => println!("RPC Connection Error: {}\n", e),
    }

    // ECDSA Signing
    println!("=== ECDSA (secp256k1) Signing ===\n");
    match EcdsaSigner::sign_transaction(PRIVATE_KEY_HEX, &tx_hex) {
        Ok(sig) => println!("{}\n", sig),
        Err(e) => println!("Signing error: {}\n", e),
    }

    Ok(())
}

// ============================================================================
// CARGO.TOML DEPENDENCIES
// ============================================================================
/*
[package]
name = "bitcoin-transaction-5.5m"
version = "0.1.0"
edition = "2021"

[dependencies]
bitcoin = "0.30"
bitcoincore-rpc = "0.17"
secp256k1 = { version = "0.27", features = ["recovery"] }
hex = "0.4"
tokio = { version = "1", features = ["full"] }
serde = { version = "1.0", features = ["derive"] }
serde_json = "1.0"

[profile.release]
opt-level = 3
lto = true
*/
